x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/action_form.tsx

```diff
@@ -240,47 +243,81 @@ export const ActionForm = ({
     }
     setIsAddActionPanelOpen(false);
     const allowGroupConnector = (actionTypeModel?.subtype ?? []).map((atm) => atm.id);
+    const isSystemAction = Boolean(
+      actionTypesIndex && actionTypesIndex[actionTypeModel.id]?.isSystemActionType
+    );
     let actionTypeConnectors = connectors.filter(
       (field) => field.actionTypeId === actionTypeModel.id
     );
-
     if (actionTypeConnectors.length > 0) {
-      actions.push({
-        id: '',
-        actionTypeId: actionTypeModel.id,
-        group: defaultActionGroupId,
-        params: {},
-        frequency: defaultRuleFrequency,
-        uuid: uuidv4(),
-      });
+      actions.push(
+        isSystemAction
+          ? {
+              id: '',
+              actionTypeId: actionTypeModel.id,
+              params: {},
+              uuid: uuidv4(),
+              type: RuleActionTypes.SYSTEM,
+            }
+          : {
+              id: '',
+              actionTypeId: actionTypeModel.id,
+              group: defaultActionGroupId,
+              params: {},
+              frequency: defaultRuleFrequency,
+              uuid: uuidv4(),
+              type: RuleActionTypes.DEFAULT,
+            }
+      );
       setActionIdByIndex(actionTypeConnectors[0].id, actions.length - 1);
     } else {
       actionTypeConnectors = connectors.filter((field) =>
         allowGroupConnector.includes(field.actionTypeId)
       );
       if (actionTypeConnectors.length > 0) {
-        actions.push({
-          id: '',
-          actionTypeId: actionTypeConnectors[0].actionTypeId,
-          group: defaultActionGroupId,
-          params: {},
-          frequency: DEFAULT_FREQUENCY,
-          uuid: uuidv4(),
-        });
+        actions.push(
```

Comment:
can we create a function to avoid duplication code?

Reply:
Pushing a simplified version of this control flow

I looked into the git blame to find out why `DEFAULT_FREQUENCY` was being set sometimes, and `defaultRuleFrequency` was being set other times, and it turns out it's due to a bad merge, so I'm just setting `defaultRuleFrequency` no matter what going forward. This may end up fixing a bug that we haven't yet detected.